testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/QueryExecutors.java

```diff
@@ -122,13 +122,21 @@ public static QueryExecutor onDelta()
         // but Spark is still initializing. It's also possible an OOM on Spark will restart it, and the gateway may
         // return 502 then as well. Handling this with a query retry allows us to use the cluster's autostart feature safely,
         // while keeping costs to a minimum.
-
-        RetryPolicy<QueryResult> databricksRetryPolicy = new RetryPolicy<QueryResult>()
+        RetryPolicy<QueryResult> databricks502RetryPolicy = new RetryPolicy<QueryResult>()
                 .handleIf(throwable -> throwable.getMessage().contains("HTTP Response code: 502"))
                 .withBackoff(1, 10, ChronoUnit.SECONDS)
                 .withMaxRetries(60)
                 .onRetry(event -> log.warn(event.getLastFailure(), "Query failed on attempt %d, will retry.", event.getAttemptCount()));
 
+        // Databricks JDBC driver retries operations that receive HTTP 503 responses
+        // if the server response is returned WITH Retry-After headers by default.
+        // Following policy retries only once when received 'HTTP retry after response' WITHOUT Retry-After headers.
+        RetryPolicy<QueryResult> databricks503RetryPolicy = new RetryPolicy<QueryResult>()
```

Comment:
Did you happen to spot this issue several times in the builds?

Reply:
It's 1 failure per 1 CI job as far as I confirmed. 
* [TestHiveAndDeltaLakeRedirect.testViewReferencingHiveAndDeltaTable](https://github.com/trinodb/trino/actions/runs/3155165327/jobs/5134503312)
* [TestDeltaLakeWriteDatabricksCompatibility.testCaseUpdateInPartition](https://github.com/trinodb/trino/actions/runs/3155158482/jobs/5134436064)
* [TestDeltaLakeDatabricksCreateTableAsSelectCompatibility.testCreateFromTrinoWithDefaultPartitionValues](https://github.com/trinodb/trino/actions/runs/3154388563/jobs/5133165509)